tsfc/kernel_interface/firedrake.py

```diff
@@ -285,9 +301,16 @@ def prepare_coefficient(coefficient, name, interior_facet=False):
     shape = finat_element.index_shape
     size = numpy.prod(shape, dtype=int)
 
-    funarg = coffee.Decl(SCALAR_TYPE, coffee.Symbol(name),
-                         pointers=[("restrict",)],
-                         qualifiers=["const"])
+# THIS IS MISSING THE OLD TENSOR_SHAPE ARGUMENT FOR THE RANK
+# IS THIS OK? ONLY MATTERS FOR TENSOR FINITE ELEMENTS I THINK?
+    if not interior_facet:
+        funarg = coffee.Decl(SCALAR_TYPE, coffee.Symbol(name, rank=(size,)), qualifiers=["const"])
+    else:
+        funarg = coffee.Decl(SCALAR_TYPE, coffee.Symbol(name, rank=(2*size,)), qualifiers=["const"])
+
+#    funarg = coffee.Decl(SCALAR_TYPE, coffee.Symbol(name),
+#                        pointers=[("restrict",)],
+#                         qualifiers=["const"])
```

Comment:
This changes the calling signature from `double *foo` to `double foo[size]`, why?